Kernel/Arch/x86_64/ISABus/I8042Controller.h

```diff
@@ -94,7 +94,11 @@ class I8042Controller final
 public:
     static ErrorOr<NonnullRefPtr<I8042Controller>> create();
 
-    ErrorOr<void> detect_devices();
+    enum class EnableFirstPortTranslation {
+        Yes,
+        No
+    };
```

Comment:
"first port" translation sounds a bit odd to me. If the point of #18654 was to remove assumptions about what's where on the serial bus, this would contradict that. How about we directly query the command line in the `KeyboardDevice` constructor?

Reply:
The problem is that PS2 set2->set1 scan code translation can enabled on the first port on the i8042 - this is not something about semantics - it's literally how the i8042 controller works. if you connect the keyboard to the second port, technically everything should work, but you can't have scan codes being translated from set 2 (if the keyboard sends scan code set 2) to set 1. if you connect a mouse to the first port and enable translation, you will get garbage data.